### ISSUE TITLE
Data as string is not queriable

### ISSUE THREAD

**Issue by behrad** [par6n/chabok-job-offer#1]
Consider a purchase event with this data:
```js
{
  productId: '123abc',
  revenue: 2000
}
```
We can't query on purchases with revenue more than 1000 in your schema!

**Comment by par6n** [comment]
Is each Event Data is consisted of a pre-set keys?

**Comment by behrad** [comment]
Exactly, Event nested data schema should be dynamic and is different per customer!

**Comment by behrad** [comment]
I will explain the data model for you
